### PR TITLE
fix matching logic for annotations and fix options metadata

### PR DESCRIPTION
**Pull request by KennethNickles**
dex test parser added fully qualified annotations, I also fixed the options annotation metadata by removing the option name so that it just uses the field name

### CODE REVIEW

File: torque-core/src/test/kotlin/com/workday/torque/ModuleTestParserSpec.kt [hunk 1]
```diff
@@ -98,7 +98,7 @@ class ModuleTestParserSpec : Spek(
 			val args = Args().apply {
 				testApkPaths = listOf(testApkPath)
 				appApkPath = targetApkPath
-				includedAnnotations = listOf("MediumTest")
+				includedAnnotations = listOf("androidx.test.filters.MediumTest")
```
Comment (JoelWilcox) [inline torque-core/src/test/kotlin/com/workday/torque/ModuleTestParserSpec.kt:101]: Do all annotations passed to Torque need to be fully qualified now? If so, we should make that explicit in all of the documentation including `Args`, `TorqueRunTask`, and also update the sample in the README

Comment (KennethNickles) [reply torque-core/src/test/kotlin/com/workday/torque/ModuleTestParserSpec.kt:101]: Yes dex test parser returns full qualified annotations now so we need to match on the full path. I'll update the documentation

File: torque-core/src/main/kotlin/com/workday/torque/ModuleTestParser.kt [hunk 2]
```diff
@@ -30,12 +31,11 @@ class ModuleTestParser(private val args: Args, private val apkTestParser: ApkTes
 			includedAnnotations: List<String>,
 			excludedAnnotations: List<String>
     ): Boolean {
-        return annotations.all { includedAnnotations.hasAnnotation(it.name) }
-                && excludedAnnotations.all { includedAnnotations.hasAnnotation(it).not() }
+        return includedAnnotations.all { annotations.hasAnnotation(it) } && excludedAnnotations.all { annotations.hasAnnotation(it).not() }
```
Comment (JoelWilcox) [inline torque-core/src/main/kotlin/com/workday/torque/ModuleTestParser.kt:34]: Nice find! This seems like it should've been caught by tests before. Are the unit tests not being run by CI or do we not have tests covering this particular logic?

Comment (KennethNickles) [reply torque-core/src/main/kotlin/com/workday/torque/ModuleTestParser.kt:34]: its spek, I can add a test to catch this

**Review by JoelWilcox — APPROVED** [review]
lgtm

File: torque-core/src/main/kotlin/com/workday/torque/ModuleTestParser.kt (hunk 2 above)
Comment (hujim) [reply torque-core/src/main/kotlin/com/workday/torque/ModuleTestParser.kt:34]: oh man 🙈

Comment (hujim) [reply torque-core/src/main/kotlin/com/workday/torque/ModuleTestParser.kt:34]: hmm did we never use exclusions before? Maybe it was a YAGNI situation hmm...
🙈

Comment (hujim) [reply torque-core/src/main/kotlin/com/workday/torque/ModuleTestParser.kt:34]: oh it wasn't me lol
but yeah curious why the test didn't trigger the fail

**Comment by hujim** [comment]
> lgtm, we should figure out the unit test skipping issue in another PR soon though 👀

Maybe the update to Spek 1.2 caused some issues with unit test task finding the spek tests.
https://github.com/Workday/torque/commit/6f69c699ab5f670abbe5f2579ea934fb740f4a5e#diff-d979b641bd0ea7c9da3fe113f9657636df1002652c75042cfe3b5203da064215R15
I'm not even able to trigger locally, even prior to that commit.
Maybe we should consider migrating to Spek 2 for better support for new AS and gradle support?

File: torque-core/src/main/kotlin/com/workday/torque/ModuleTestParser.kt (hunk 2 above)
Comment (KennethNickles) [reply torque-core/src/main/kotlin/com/workday/torque/ModuleTestParser.kt:34]: yeah this was my bad, the extensions actually got confusing when the types changed from dex test parser

**Comment by KennethNickles** [comment]
> Maybe the update to Spek 1.2 caused some issues with unit test task finding the spek tests.
> [6f69c69#diff-d979b641bd0ea7c9da3fe113f9657636df1002652c75042cfe3b5203da064215R15](https://github.com/Workday/torque/commit/6f69c699ab5f670abbe5f2579ea934fb740f4a5e#diff-d979b641bd0ea7c9da3fe113f9657636df1002652c75042cfe3b5203da064215R15)
> I'm not even able to trigger locally, even prior to that commit.
> Maybe we should consider migrating to Spek 2 for better support for new AS and gradle support?

I would be in favor of dropping spek, its hard to maintain

**Comment by hujim** [comment]
> > Maybe the update to Spek 1.2 caused some issues with unit test task finding the spek tests.
> > [6f69c69#diff-d979b641bd0ea7c9da3fe113f9657636df1002652c75042cfe3b5203da064215R15](https://github.com/Workday/torque/commit/6f69c699ab5f670abbe5f2579ea934fb740f4a5e#diff-d979b641bd0ea7c9da3fe113f9657636df1002652c75042cfe3b5203da064215R15)
> > I'm not even able to trigger locally, even prior to that commit.
> > Maybe we should consider migrating to Spek 2 for better support for new AS and gradle support?
> 
> I would be in favor of dropping spek, its hard to maintain

Yeah agree, but a rewrite of all the tests would be a pretty big effort.

**Comment by KennethNickles** [comment]
> Yeah agree, but a rewrite of all the tests would be a pretty big effort.

I can probably do it in a few weeks. Should have some free time

**Pull request merged**